google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/UserSpecifiedPathDeploymentSource.java

```diff
@@ -59,13 +60,19 @@ public Icon getIcon() {
   @NotNull
   @Override
   public String getPresentableName() {
-    return name;
+    return String.format("[%s] ", DateTime.now().toString("yyyy-MM-dd HH:mm:ss")) + name;
```

Comment:
I have been pondering this too. Its a little tricky given the inheritance hierarchy we are stuck with given the different deployment types. I am leaning towards saying this is ok because we can think of `presentableName` as being something unique to each individual deployment source, although I agree the duplication of this date format is less than ideal.